lightning-rapid-gossip-sync/src/processing.rs

```diff
@@ -435,6 +439,50 @@ mod tests {
 		assert!(after.contains("783241506229452801"));
 	}
 
+	#[test]
+	fn update_succeeds_when_duplicate_gossip_update_from_server() {
```

Comment:
_with_ duplicate gossip

Reply:
given-when-then-should is general unit-test naming convention, we don't follow it as such. 
but it was derived from that. happy to change if you think.

Reply:
if the grammar worked with when/then, sure, but in its current configuration it doesn't, and making it so would be very verbose

Reply:
alternatively, `update_succeeds_when_duplicate_gossip_is_applied`